### ISSUE TITLE
Option to override default value escaping/formatting 

### ISSUE THREAD

**Issue by yellowpattern** [grafana/grafana#6789]
Grafana 3.1
Datasource: InfluxDB 1.1
OS: CentOS 6

When using template variables, it can be useful for the variable to expand to a regex.

For example, if I'm looking to display network stats, I might want a list like this:

lo0, eth0, eth1, eth.*

Why not just use "multi-select"?

Because when the list of objects to "select" grows out to be more than 20, from a set of about 80 metrics, the drop downs can be rather hard to use when just using the "SHOW TAG" query with influxdb.

Why can't I use "eth.*" now?

If I have a custom variable with "lo,eth0,eth1,eth.\*" as the available choices, if I select "eth.\*" in the non-raw editor mode, the regex that gets sent to influxdb is /^eth\\.\\\*$/.

That's not very useful. Even in raw mode this cannot be worked around unless the regex is removed from the variable.

While the "/^$host$/" works for simple cases, it would be nice if there was an alternate mode for regex matching that does not include anchors (no "^" at the front, not "$" at the back) and does not quote special characters when passing it through to influxdb.

**Comment by freeseacher** [comment]
looks like mine #6179

**Comment by jeredfloyd** [comment]
See #6788 which is a related issue.

**Comment by sferrett** [comment]
This used to work for me in version 4.1.2 but broke when I upgraded to 4.3.2
I have a template variable that I use in a subsequent regex filter, and as a search in a query.  Since the new version the regex specials are backslash-quoted and so they no longer work.

eg: my template var sets $filter to be `.*` and the resulting query to influx is (observed via chrome dev tools):
`show tag values from cpu_value with key = host where host =~ /\.\*/`

So the query returns no matches.  When I used 4.1.2 this worked just fine.
